### DISCUSSION TITLE
Problems with PHP

### DISCUSSION THREAD

**Discussion by Rocketst** [community/community#46369] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

Good afternoon

I am new to this topic.
The code below should output the page content by date filter.
However, it outputs the entries  [includeme post_id="181526"] он this page https://joycemeyer.ru/promises/
Why is this happening?

```
<?
function remove_fusion_shortcodes( $content ) {
    $content = preg_replace('/\[\/?fusion.*?\]/', '', $content);    
    //$content = preg_replace('/\<.*?\>/', '', $content);
    //$content = preg_replace('/http.*?jpg/', '', $content);

    $content = mb_substr($content,mb_strpos ($content, ".jpg",mb_strpos ($content, ".jpg")+4)+5,700);
    return $content;
}


   //echo "<pre>"; print_r($query_string); echo "</pre>";
   //echo "<pre>"; var_dump($GLOBALS["wp_the_query"]); echo "</pre>";

$today = getdate();
$my_posts = new WP_Query();

//$myposts = $my_posts->query(array(
$myposts = get_posts(array(
	'date_query' => array(
		array(
			'month' => $today["mon"], // обратите внимание, раньше было monthnum, теперь просто month
			'day'   => $today["mday"],
		),
	),
    'cat' => 222, 'orderby' => 'date','order' => 'DESC','posts_per_page' => 1
) );

$arPosts = array();
foreach( $myposts as $pst ) {
    $image = wp_get_attachment_image_src(get_post_thumbnail_id($pst->ID),"full");
    $arPosts[] = array("TITLE" => esc_html( $pst->post_title ), "DESC" => (get_the_excerpt($pst->ID)?get_the_excerpt($pst->ID):strip_tags(remove_fusion_shortcodes(preg_replace("#http?://[a-zA-Z0-9_] \z#i", " ",$pst->post_content)),"<br><a>")."..."), "IMAGE" => $image[0],"DATE"=>$pst->post_date,"LINK" => get_permalink($pst->ID));
    if( count($arPosts) >= 1 ) break;
}


//if(strftime("%Y.%m.%d",strtotime($arPosts[0]["DATE"]))!=date("Y.m.d")){
if(count($arPosts)<1){
    wp_reset_postdata();
    unset($my_posts);
    $my_posts = new WP_Query();
    $myposts = $my_posts->query( array(
	'date_query' => array(
		array(
			'before'    => array(
				'year' => $today["year"],
				'month' => $today["mon"],
				'day'   => $today["mday"],
			)
		)
	), 'cat' => 222, 'orderby' => 'date','order' => 'DESC'
    ) );

    $arPosts = array();
    foreach( $myposts as $pst ) {
        $image = wp_get_attachment_image_src(get_post_thumbnail_id($pst->ID),"full");
        $arPosts[$pst->post_date] = array("TITLE" => esc_html( $pst->post_title ), "DESC" => (get_the_excerpt($pst->ID)?get_the_excerpt($pst->ID):strip_tags(preg_replace("#http://[a-zA-Z0-9_-]\s#i", " ",remove_fusion_shortcodes($pst->post_content)),"<br><a>")."..."), "IMAGE" => $image[0],"DATE"=>$pst->post_date,"LINK" => get_permalink($pst->ID));
      if( count($arPosts) >= 1 ) break;
   }
   //echo "<pre>"; print_r($arPosts); echo "</pre>";
}
$months = array( 1 => 'Января' , 'Февраля' , 'Марта' , 'Апреля' , 'Мая' , 'Июня' , 'Июля' , 'Августа' , 'Сентября' , 'Октября' , 'Ноября' , 'Декабря' );

?>
</Конец функции вывода данных из базы данных WorPress --> 

  


<div>

        </Вывод описания статьи из блока expert --> 
       <?=$arPosts[0]["DESC"]?>

</div>


    
    
<?   
unset($my_posts);
wp_reset_postdata();
?>
```

**Comment by jge162** [comment]
Hello @Rocketst issue could be due to a conflict between the shortcode and the website's theme or plugin. To troubleshoot the issue, you can try disabling all plugins and switching to a default WordPress theme, and see if the issue persists. If it does, you can try checking the code of the shortcode for any syntax errors or conflicting functions.

**Reply by Rocketst** [reply]
The Include Me plugin was actively used on my site - https://ru.wordpress.org/plugins/include-me/
In particular, the shortcode [include post id = "post id"] for displaying content[ on this page](https://joycemeyer.ru/promises/)

[This page](https://joycemeyer.ru/promises/) uses php (the code above) which should display the post by publication date, date to date. For example, a post published on February 18 will appear on this page on February 18. And so every day, the content changes depending on the date.

In the new version of WP, the Include Me plugin stopped working.
Since the failure of the Include Me plugin, the php (the code above) has been deduced by the WPCode plugin.

However, instead of the content of the page of the desired day (for example, [this page for February 18](https://joycemeyer.ru/pr-18-02/)), the shortcode [include post id = "post id"] is displayed. "Post ID" changes according to the day.

As I understand it, the php code (below) generates the "post id" for the shortcode [include post id = "post id"] and displays them [on the page](https://joycemeyer.ru/promises/).

The question is how to finish everything so that the content of the page is shown, and not the shortcode of the non-working plugin [include post id = "post id"] ???
